nixos/modules/services/wayland/cage.nix

```diff
@@ -0,0 +1,82 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cage;
+in {
+  options.services.cage.enable = mkEnableOption ''
+    Enable the Cage Kiosk service.
+  '';
+
+  options.services.cage.user = mkOption {
+    type = types.str;
+    default = "demo";
+    description = ''
+      User to log-in as.
+    '';
+  };
+
+  options.services.cage.program = mkOption {
+    type = types.path;
+    default = "${pkgs.xterm}/bin/xterm";
+    description = ''
+      Program to run in cage.
+    '';
+  };
+
+  config = mkIf cfg.enable {
+
+    # The service is partially based off of the one provided in the
+    # cage wiki at
+    # https://github.com/Hjdskes/cage/wiki/Starting-Cage-on-boot-with-systemd.
+    systemd.services."cage@" = {
+      enable = true;
+      after = [ "systemd-user-sessions.service" "plymouth-quit.service" "dbus.socket" "systemd-logind.service" "getty@%i.service" "plymouth-deactivate.service" ];
+      before = [ "graphical.target" ];
+      wants = [ "dbus.socket" "systemd-logind.service" "plymouth-deactivate.service" ];
+      wantedBy = [ "graphical.target" ];
+      conflicts = [ "getty@%i.service" "plymouth-quit.service" "plymouth-quit-wait.service" ];
+
+      restartIfChanged = false;
+      serviceConfig = {
+        ExecStart = "${pkgs.cage}/bin/cage -d -- ${cfg.program}";
```

Comment:
`cage` supports rotation: https://github.com/Hjdskes/cage/pull/57

> This commit adds two command line switches. -r rotates the output 90
degrees clockwise and can be specified multiple times. -f flips the
output.

(We might need to bump the nixpkgs version for that, not sure - I checked with latest master)

Reply:
yeah perhaps just some "extraArgs" or something like it.

Reply:
Adressed in  #81327.